### PR TITLE
Deprecate with @GraphQLDeprecated

### LINKED ISSUE #1492: @Deprecated, but only for the client
image

**Is your feature request related to a problem? Please describe.**

It's cool that graphql-kotlin supports `@Deprecated` out of the box.

But often I want to mark something as deprecated for my clients
but not on my own code since I'm still implementing the feature.

Since I'm converting all my warnings in errors, I have to add `@Suppress("DEPRECATION")` everywhere.

**Describe the solution you'd like**

Add a `@GraphQLDeprecated("message")` annotation that would be sent to the client but not pollute my own code.

**Describe alternatives you've considered**

For the naming I chose something similar to `@GraphQLDocumentation("documentation")`

`@file:Suppress("DEPRECATION")` everywhere relevant but then I'm loosing the benefits of Kotlin's `@Deprecated` annotation

I could wire up my own annotation processor but it would be cool it worked out of the box just like `@Deprecated`

### PR DESCRIPTION
**Pull request by jmfayard**
### :pencil: Description

Works the same than @Depcreated (which is still fine and supported)
but doesn't deprecate your own code, only deprecate in the client

### :link: Related Issues

Resolves #1492

### CODE REVIEW

File: website/docs/schema-generator/customizing-schemas/annotations.md [hunk 1]
```diff
@@ -10,4 +10,5 @@ for things that can't be directly derived from Kotlin reflection.
 - [@GraphQLIgnore](./excluding-fields) - Exclude field from the GraphQL schema
 - [@GraphQLName](./renaming-fields) - Override the name used for the type
 - Kotlin built in [@Deprecated](./deprecating-schema) - Apply the GraphQL `@deprecated` directive on the field
+- [@GraphQLDeprecated](./deprecating-schema) - Same but when you don't want to deprecated only for the client, not in your own code
```
Comment (smyrick) [inline website/docs/schema-generator/customizing-schemas/annotations.md:13]:
```suggestion
- [@GraphQLDeprecated](./deprecating-schema) - Apply the GraphQL `@deprecated` directive but only in the schema, not in your own Kotlin code with `@Deprecated`
```

File: website/docs/schema-generator/customizing-schemas/deprecating-schema.md [hunk 2]
```diff
@@ -25,7 +33,22 @@ type Query {
...
+
+
+```kotlin
+class SimpleQuery {
+  @GraphQLDeprecated(message = "this query is deprecated", replaceWith = ReplaceWith("shinyNewQuery"))
+  fun simpleDeprecatedQuery(): Boolean = false
+
+  fun shinyNewQuery(): Boolean = true
+}
+```
```
Comment (smyrick) [inline website/docs/schema-generator/customizing-schemas/deprecating-schema.md:54]:
```suggestion
A side-effect of using `@Deprecated` is that it marks your own Kotlin code as being deprecated, which may not be what you want. Using `@GraphQLDeprecated` you can add the `@deprecated` directive to the GraphQL schema, but not have your Kotlin code show up as deprecated in your editor.

```kotlin
class SimpleQuery {
  @GraphQLDeprecated(message = "this query is deprecated", replaceWith = ReplaceWith("shinyNewQuery"))
  fun simpleDeprecatedQuery(): Boolean = false

  fun shinyNewQuery(): Boolean = true
}
```
```

**Review by smyrick — CHANGES_REQUESTED** [review]

**Comment by EXPErdelfavero** [comment]
Changes to the Markdown here look good from a grammar/style standpoint. (Cheers to @smyrick who got everything before I saw this 😸...) I don't comment on code changes or the code snippets inside the Markdown files, though. I leave that to the experts...

File: generator/graphql-kotlin-schema-generator/src/test/kotlin/com/expediagroup/graphql/generator/internal/extensions/AnnotationExtensionsTest.kt [hunk 3]
```diff
@@ -54,6 +55,14 @@ class AnnotationExtensionsTest {
     @GraphQLUnion(name = "MetaUnion", possibleTypes = [NoAnnotations::class])
     annotation class MetaUnion
 
+    @GraphQLDeprecated("class deprecated", ReplaceWith("WithAnnotations"))
+    @GraphQLIgnore
```
Comment (dariuszkuc) [inline generator/graphql-kotlin-schema-generator/src/test/kotlin/com/expediagroup/graphql/generator/internal/extensions/AnnotationExtensionsTest.kt:59]: is the ignore intentional?

Comment (jmfayard) [reply generator/graphql-kotlin-schema-generator/src/test/kotlin/com/expediagroup/graphql/generator/internal/extensions/AnnotationExtensionsTest.kt]: No, copy/pasted it from the original example. I removed it.

**Review by dariuszkuc — APPROVED** [review]
Awesome work!

**Review by smyrick — APPROVED** [review]

**Pull request merged**